Samples/src/OracleEmulation/Operations.qs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+namespace Microsoft.Quantum.Samples.OracleEmulation
+{
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Primitive;
+    open Microsoft.Quantum.Extensions.Emulation;
+
+
+    //////////////////////////////////////////////////////////////////////////
+    // Defining and using simple emulated oracles ////////////////////////////
+    //////////////////////////////////////////////////////////////////////////
+
+    // Declare an `intrinsic` oracle that is implemented in C#. See the
+    // `PermutationOracle.Register` call in the driver for its implementation.
+    operation HalfAnswer(x: Qubit[], y: Qubit[]) : Unit {
+        body intrinsic;
+        adjoint intrinsic;
+    }
+
+    // Define a simple permutation function that is used below to create
+    // another oracle.
+    function DoubleAnswerFunc(x: Int, y: Int) : Int {
+        return 84 ^^^ y;
+    }
+
+    // Measure and print the result.
+    operation MeasureAndDisplay(message: String, register: Qubit[]) : Unit {
+        let answer = MeasureInteger(LittleEndian(register));
+        Message(message + $"{answer}.");
+    }
+
+    // # Summary
+    // Here we demonstrate the use of three simple oracles. Each oracle ignores
+    // the content of the first register and XOR's a constant number into the
+    // second register.
+    //
+    // # Input
+    // ## oracle
+    // A quantum operation that implements an oracle
+    //      $O: \ket{x}\ket{y} \rightarrow \ket{x}\ket{f(x, y)}$.
+    operation RunConstantOracles (oracle: ((Qubit[], Qubit[]) => Unit)) : Unit {
+        Message("Querying the oracles...");
+
+        // Prepare a one-qubit register `x` and an eight-qubit register `y`.
+        // Since all the oracles here ignore x, its length and state do not
+        // matter.
+        using (qubits = Qubit[9]) {
```

Comment:
Generally, it's easier to read and maintain to allocate qubit variables directly using tuple deconstruction:
```suggestion
        using ((x, y) = (Qubit(), Qubit[8])) {
```
I'd also suggest more descriptive names for each register as well.

Reply:
Good ideas, done!